### PR TITLE
[#63] As a user, I can get other author profile information

### LINKED ISSUE #63: As a user, I can get other author profile information
## Why
The users will be able to get other user profiles from server.

## Acceptance Criteria
- [ ] Implement the API for getting other user profile via this [endpoint](https://github.com/gothinkster/realworld/tree/master/api#get-profile).
- [ ] The API request takes `username` as the query parameter, no need to use request body.
- [ ] Create a usecase for getting other user profile for later usage in integrate task.
- [ ] Parse the response into the corresponding model, here is an example response for getting other user profile:
```
{
  "profile": {
    "username": "jake",
    "bio": "I work at statefarm",
    "image": "https://static.productionready.io/images/smiley-cyrus.jpg",
    "following": false
  }
}
```

### PR DESCRIPTION
**Pull request by minhnimble**
Resolved #63

## What happened

The users will be able to get other user profiles from server.

## Insight

- [x] Implement the API for getting other user profile via this [endpoint](https://github.com/gothinkster/realworld/tree/master/api#get-profile).
- [x] The API request takes `username` as the query parameter, no need to use request body.
- [x] Create a usecase for getting other user profile for later usage in integrate task.
- [x] Parse the response into the corresponding model.
- [x] Added some code refactor for folders and the new use case sorting.
- [x] Added unit tests for the above use case.

## Proof Of Work

Success response log:

```
200 'https://conduit.productionready.io/api/profiles/Gerome' [0.9155 s]:
Headers: [
  report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=TKlKhMrUCK4LdrFmR9r7SG5CqlV8CRw9m42JJAUkNaMfJiSK7nLEN%2Bs%2BwNf%2BqkSzGmKqmb1CKLhmOGos9x0NMNmb38x9EWwZbEUZynOEFFBZABxIsLDaGS5Gwa%2F06zk1NVLUEw6v2veRh9u%2BOz%2F6"}],"group":"cf-nel","max_age":604800}
  cf-ray: 6953f6ac1d6d3d9a-HKG
  Content-Type: application/json; charset=utf-8
  x-powered-by: Express
  expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
  Etag: W/"4b-fY/oSly4M7RCJERNXILsmwyJXps"
  Via: 1.1 vegur
  Content-Encoding: br
  nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
  Access-Control-Allow-Origin: *
  Date: Mon, 27 Sep 2021 10:14:11 GMT
  Alt-Svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400, h3-28=":443"; ma=86400, h3-27=":443"; ma=86400
  Server: cloudflare
  cf-cache-status: DYNAMIC
]
Body:
{
  "profile" : {
    "username" : "Gerome",
    "image" : null,
    "following" : false,
    "bio" : null
  }
}
```

Tests run successfully for the new use case:

Screen Shot 2021-09-27 at 16 54 54

Coverage:

Screen Shot 2021-09-27 at 16 54 20

### CODE REVIEW

File: NimbleMedium/Sources/Data/NetworkAPI/RequestConfigurations/UserRequestConfiguration.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        switch self {
+        case .profile:
+            return .get
+        }
+    }
+
+    var parameters: Parameters? {
+        switch self {
+        case .profile:
+            return [:]
```
Comment (markgravity) [inline NimbleMedium/Sources/Data/NetworkAPI/RequestConfigurations/UserRequestConfiguration.swift:36]: We can return `nil` here 👍

File: NimbleMedium/Sources/Data/NetworkAPI/RequestConfigurations/UserRequestConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    }
+
+    var encoding: ParameterEncoding { URLEncoding.queryString }
+
+    var headers: HTTPHeaders? { nil }
+
+    var interceptor: RequestInterceptor? { nil }
```
Comment (markgravity) [inline NimbleMedium/Sources/Data/NetworkAPI/RequestConfigurations/UserRequestConfiguration.swift:44]: We can remove these cause we already added default values by extension 👍

File: NimbleMedium/Sources/Domain/UseCases/Article/GetListArticlesUseCase.swift [hunk 3]
```diff
@@ -1,5 +1,5 @@
 //
-//  ListArticlesUseCase.swift
+//  GetListArticlesUseCase.swift
```
Comment (suho) [inline NimbleMedium/Sources/Domain/UseCases/Article/GetListArticlesUseCase.swift:2]: Please separate into a chore PR to refactor. This pull request should not contains un-related code.

**Review by markgravity — APPROVED** [review]
LGTM

**Review by suho — APPROVED** [review]

**Pull request merged**